### PR TITLE
[DOCS] Create Beaconing Detections page (experimental features)

### PR DESCRIPTION
**Pull request by benironside**
Addresses issue [1282](https://github.com/elastic/security-docs/issues/1282)

[Preview](https://security-docs_1349.docs-preview.app.elstc.co/guide/en/security/master/network-beaconing-framework.html)

### CODE REVIEW

File: docs/experimental-features/beaconing-detection.asciidoc [hunk 1]
```diff
@@ -0,0 +1,50 @@
+[[beaconing-detection]]
+== Beaconing detection
+
+This feature provides an early warning system for command and control (C2) beaconing activity. It monitors network traffic for indicators of compromise (IOCs), and provides analytics to aid your threat hunting.
```
Comment (joepeeples) [inline docs/experimental-features/beaconing-detection.asciidoc:4]: Small grammar fix, but also: if the abbreviations "C2" and "IOCs" aren't used again on this page, do they need to be abbreviated at all?
```suggestion
This feature provides an early warning system for command and control (C2) beaconing activity. It monitors network traffic for indicators of compromise (IOCs) and provides analytics to aid your threat hunting.
```

File: docs/experimental-features/beaconing-detection.asciidoc [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+[discrete]
+=== Deploy the Beaconing detection package
+
+To deploy Beaconing detection in your environment, follow https://github.com/elastic/detection-rules/blob/main/docs/experimental-machine-learning/beaconing.md[these steps].
```
Comment (joepeeples) [inline docs/experimental-features/beaconing-detection.asciidoc:9]:
```suggestion
=== Deploy the beaconing detection package

To deploy beaconing detection in your environment, follow https://github.com/elastic/detection-rules/blob/main/docs/experimental-machine-learning/beaconing.md[these steps].
```

File: docs/experimental-features/beaconing-detection.asciidoc [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+[role="screenshot"]
+image::images/beaconing-detection-1.png[]
+
+[discrete]
+=== Feature details
+
+The feature uses a {ref}/transforms.html[transform] to segment network data by host and process name, then runs scripted metric aggregations on the host-process name pairs. For a given time window, the scripted metric aggregation checks each partition for the following:
+
+* Periodicity of signals, accounting for jitter.
```
Comment (joepeeples) [inline docs/experimental-features/beaconing-detection.asciidoc:21]: This seems pretty jargon-heavy. Some cybersecurity audiences might be familiar with "periodicity" and "jitter," but probably not general readers.

File: docs/experimental-features/beaconing-detection.asciidoc [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+* Low variation of bytes sent from source to destination.
+* Low variation of bytes sent from destination to source.
+
+The transform also filters out trusted applications and IPs in an attempt to reduce false positives. The transform outputs information about the process and host indicators, and the detection. It runs every hour to poll for new beacons.
```
Comment (joepeeples) [inline docs/experimental-features/beaconing-detection.asciidoc:25]: Slight revision to avoid the awkward comma. Also, is this referring to the [trusted applications](https://www.elastic.co/guide/en/security/current/trusted-apps-ov.html) feature in Elastic Security? If it's a separate thing, maybe use a different term to avoid confusion with the feature.
```suggestion
The transform also filters out trusted applications and IPs in an attempt to reduce false positives. The transform outputs information about the detection and the process and host indicators. It runs every hour to poll for new beacons.
```

File: docs/experimental-features/beaconing-detection.asciidoc [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+
+[discrete]
+=== Further customizations
+
+Advanced users can also tune the scripted metric aggregation's parameters, such as jitter percentage or time window. To overwrite the default parameters, delete the transform, change the parameters, and restart. The configurable parameters are as follows:
```
Comment (joepeeples) [inline docs/experimental-features/beaconing-detection.asciidoc:30]: Unclear what the user needs to "restart" here -- the transform? an instance of Elasticsearch or Kibana?
```suggestion
Advanced users can also tune the scripted metric aggregation's parameters, such as jitter percentage or time window. To overwrite the default parameters: delete the transform, change the parameters, and restart. The configurable parameters are:
```

File: docs/experimental-features/beaconing-detection.asciidoc [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+* `number_buckets_in_range:`
+* `time_bucket_length:`
+* `number_destination_ips:`
+* `truncate_at:`
+* `max_beaconing_bytes_cov:`
+* `max_beaconing_count_rv:`
+* `min_beaconing_count_autocovariance:`
+* `min_number_periods:`
+* `min_not_empty:`
+* `max_jitter:`
```
Comment (joepeeples) [inline docs/experimental-features/beaconing-detection.asciidoc:41]: An explanation for each parameter would be helpful, especially for some that don't seem immediately self-explanatory. Not sure this is a deal-breaker for an experimental feature, though, where there might be less expectation of exhaustive documentation -- @jmikell821 WDYT?

File: docs/experimental-features/beaconing-detection.asciidoc [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+
+You can also make changes to the transform query. The default query looks for beaconing activity over a 6h time range, but you can change this to a different time range. As mentioned previously, beaconing is not a characteristic specific to malware and a lot of legitimate, benign processes also exhibit beacon-like activity. In order to reduce false positives, a list of starter filters are included by default in the transform query to exclude known beaconing processes, and IPs that fall into two groups:
+
+* The source IP is local and the destination is remote.
+* For certain Microsoft processes, the destination IP is in a Microsoft block.
+
+You can add to this list based on your environment.
+
+We are working towards embedding this feature in {elastic-sec}.
```
Comment (joepeeples) [inline docs/experimental-features/beaconing-detection.asciidoc:50]: IMO this seems unnecessary, as this is true for any experimental or beta feature. The [landing page](https://www.elastic.co/guide/en/security/current/sec-experimental-intro.html) for the Experimental features section kind of covers this too.

File: docs/experimental-features/beaconing-detection.asciidoc [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+* `number_destination_ips:`
+* `truncate_at:`
+* `max_beaconing_bytes_cov:`
+* `max_beaconing_count_rv:`
+* `min_beaconing_count_autocovariance:`
+* `min_number_periods:`
+* `min_not_empty:`
+* `max_jitter:`
+
+You can also make changes to the transform query. The default query looks for beaconing activity over a 6h time range, but you can change this to a different time range. As mentioned previously, beaconing is not a characteristic specific to malware and a lot of legitimate, benign processes also exhibit beacon-like activity. In order to reduce false positives, a list of starter filters are included by default in the transform query to exclude known beaconing processes, and IPs that fall into two groups:
```
Comment (joepeeples) [inline docs/experimental-features/beaconing-detection.asciidoc:43]: Also consider starting a new paragraph with "As mentioned previously" since the topic shifts there. (I'd add it myself but it goofs up the diff view.)
```suggestion
You can also make changes to the transform query. The default query looks for beaconing activity over a 6-hour time range, but you can change this to a different time range. As mentioned previously, beaconing is not specific to malware, and a lot of legitimate, benign processes also exhibit beacon-like activity. In order to reduce false positives, a list of starter filters is included by default in the transform query to exclude known beaconing processes and IPs that fall into two groups:
```

File: docs/experimental-features/beaconing-detection.asciidoc [hunk 9]
```diff
@@ -0,0 +1,50 @@
...
+
+* The source IP is local and the destination is remote.
+* For certain Microsoft processes, the destination IP is in a Microsoft block.
```
Comment (joepeeples) [inline docs/experimental-features/beaconing-detection.asciidoc:46]: Might need to explain what a "Microsoft block" is, or maybe I'm just not familiar with the term?

**Review by joepeeples — COMMENTED** [review]
A few suggested revisions. Let me know if you have any questions!

File: docs/experimental-features/beaconing-detection.asciidoc (hunk 9 above)
Comment (benironside) [reply docs/experimental-features/beaconing-detection.asciidoc:46]: I think it's referring to a block of IP addresses that Microsoft owns/uses

File: docs/experimental-features/beaconing-detection.asciidoc (hunk 3 above)
Comment (benironside) [reply docs/experimental-features/beaconing-detection.asciidoc:21]: I think that periodicity of signals = signals repeating at regular intervals, jitter = minor variations in those regular intervals. I feel like a user to look up the terms if they were unfamiliar, but we could also say "Signals repeating at regular intervals, accounting for minor variations in those intervals." Do you think that would be better @joepeeples

Comment (joepeeples) [reply docs/experimental-features/beaconing-detection.asciidoc:21]: @benironside Yes, that's a lot more readable to me, thanks!

File: docs/experimental-features/beaconing-detection.asciidoc (hunk 9 above)
Comment (joepeeples) [reply docs/experimental-features/beaconing-detection.asciidoc:46]: @benironside That makes sense, especially in this context. At first I read "block" as meaning "something being banned," not "a block of (allowed?) IP addresses," so maybe rephrase to something like you just said ("a block of IP addresses...").

Comment (benironside) [reply docs/experimental-features/beaconing-detection.asciidoc:46]: 💯

File: docs/experimental-features/beaconing-detection.asciidoc (hunk 6 above)
Comment (benironside) [reply docs/experimental-features/beaconing-detection.asciidoc:41]: I'll reach out to the dev for more detail here.

File: docs/experimental-features/beaconing-detection.asciidoc (hunk 5 above)
Comment (benironside) [reply docs/experimental-features/beaconing-detection.asciidoc:30]: Good question, I'll clarify with the dev.

File: docs/experimental-features/beaconing-detection.asciidoc [hunk 10]
```diff
@@ -0,0 +1,49 @@
...
+[role="screenshot"]
+image::images/beaconing-detection-2.png[]
+
+[discrete]
+=== Further customizations
+
+Advanced users can also tune the scripted metric aggregation's parameters, such as jitter percentage or time window. To overwrite the default parameters: delete the transform, change the parameters, and restart the transform. The configurable parameters are:
+
+* `number_buckets_in_range:` The number of time buckets into which the time window is split. Using more buckets will improve estimates for the various statistics, but also increase resource usage.
+* `time_bucket_length:` The length of each time bucket. The larger this value the longer the time window. Set this longer to check for very low frequency beacons.
```
[the remark this answers is not in the document]
Comment (joepeeples) [reply docs/experimental-features/beaconing-detection.asciidoc:36]: ^^ +1, and also revise to "low-frequency" since it's acting as an adjective modifying "beacons."

File: docs/experimental-features/beaconing-detection.asciidoc [hunk 11]
```diff
@@ -0,0 +1,49 @@
...
+
+[role="screenshot"]
+image::images/beaconing-detection-2.png[]
+
+[discrete]
+=== Further customizations
+
+Advanced users can also tune the scripted metric aggregation's parameters, such as jitter percentage or time window. To overwrite the default parameters: delete the transform, change the parameters, and restart the transform. The configurable parameters are:
+
+* `number_buckets_in_range:` The number of time buckets into which the time window is split. Using more buckets will improve estimates for the various statistics, but also increase resource usage.
```
Comment (joepeeples) [inline docs/experimental-features/beaconing-detection.asciidoc:35]: Very small detail, but I'd put the colon outside the monospace backticks, so it doesn't look like part of the setting name. (I won't add the code suggestion to every line since that'd be noisy, just this first instance to illustrate how the others should also look.)
```suggestion
* `number_buckets_in_range`: The number of time buckets into which the time window is split. Using more buckets will improve estimates for the various statistics, but also increase resource usage.
```

File: docs/experimental-features/beaconing-detection.asciidoc [hunk 12]
```diff
@@ -0,0 +1,49 @@
...
+* `max_beaconing_bytes_cov:` The maximum coefficient of variation in the payload bytes for the low source and destination bytes variance test. Higher values increase the chance of flagging traffic as beaconing, increasing https://en.wikipedia.org/wiki/Precision_and_recall[recall] while reducing https://en.wikipedia.org/wiki/Precision_and_recall[precision].
+* `max_beaconing_count_rv:` The maximum relative variance in the bucket counts for the high frequency beacon test. As with `max_beaconing_bytes_cov`, tuning this parameter involves a tradeoff between recall and precision.
+* `truncate_at`: The lower and upper fraction of bucket values discarded when computing `max_beaconing_bytes_cov` and `max_beaconing_count_rv`. This allows you to ignore occasional changes in traffic patterns. However, if you retain too small a fraction of the data, these tests will be unreliable.
+* `min_beaconing_count_autocovariance:` The minimum autocorrelation of the signal for the low frequency beacon test. Lowering this value generally increases recall for malicious C2 beacons, while reducing precision.
```
Comment (joepeeples) [inline docs/experimental-features/beaconing-detection.asciidoc:41]:
```suggestion
* `min_beaconing_count_autocovariance:` The minimum autocorrelation of the signal for the low-frequency beacon test. Lowering this value generally increases recall for malicious command and control beacons, while reducing precision.
```

File: docs/experimental-features/beaconing-detection.asciidoc [hunk 13]
```diff
@@ -0,0 +1,49 @@
...
+* `max_jitter:` The maximum amount of https://en.wikipedia.org/wiki/Jitter[jitter] assumed to be possible for a periodic beacon, as a fraction of its period.
+
+You can also make changes to the transform query. The default query looks for beaconing activity over a 6-hour time range, but you can use a different time range.
+
+As mentioned previously, beaconing is not used exclusively by malware. Many legitimate, benign processes also exhibit beacon-like activity. In order to reduce false positives, a list of starter filters is included by default in the transform query to exclude known beaconing processes, and IPs that fall into two groups:
```
Comment (joepeeples) [inline docs/experimental-features/beaconing-detection.asciidoc:46]:
```suggestion
As mentioned previously, beaconing is not used exclusively by malware. Many legitimate, benign processes also exhibit beacon-like activity. In order to reduce false positives, a list of starter filters is included by default in the transform query to exclude known beaconing processes and IPs that fall into two groups:
```

**Review by joepeeples — COMMENTED** [review]
A few more suggestions, looks good!

File: docs/experimental-features/beaconing-detection.asciidoc [hunk 14]
```diff
@@ -0,0 +1,49 @@
...
+
+The transform, which runs every hour, also filters out trusted applications and IPs in an attempt to reduce false positives. The transform outputs information about the detection and the process and host indicators, for example: 
+
+[role="screenshot"]
+image::images/beaconing-detection-2.png[]
+
+[discrete]
+=== Further customizations
+
+Advanced users can also tune the scripted metric aggregation's parameters, such as jitter percentage or time window. To overwrite the default parameters: delete the transform, change the parameters, and restart the transform. The configurable parameters are:
```
[the remark this answers is not in the document]
Comment (benironside) [reply docs/experimental-features/beaconing-detection.asciidoc:33]: haha, I had a comma to start and Joe suggested a colon, which I like.

File: docs/experimental-features/beaconing-detection.asciidoc [hunk 15]
```diff
@@ -0,0 +1,49 @@
...
+The feature uses a {ref}/transforms.html[transform] to segment network data by host and process name, then runs scripted metric aggregations on the host-process name pairs. For a given time window, the scripted metric aggregation checks each partition for the following:
+
+* Signals repeating at regular intervals, accounting for minor variations in those intervals.
+* Low variation of bytes sent from source to destination.
+* Low variation of bytes sent from destination to source.
+
+The transform, which runs every hour, also filters out trusted applications and IPs in an attempt to reduce false positives. The transform outputs information about the detection and the process and host indicators, for example: 
+
+[role="screenshot"]
+image::images/beaconing-detection-2.png[]
```
[the remark this answers is not in the document]
Comment (benironside) [reply docs/experimental-features/beaconing-detection.asciidoc:28]: Fine with me — I'll have to get the non-annotated version from Sourin or Apoorva since they provided it with the annotations.

File: docs/experimental-features/beaconing-detection.asciidoc (hunk 10 above)
[the remark this answers is not in the document]
Comment (benironside) [reply docs/experimental-features/beaconing-detection.asciidoc:36]: Great suggestions, thank you both 🔥 🔥 🔥

File: docs/experimental-features/beaconing-detection.asciidoc (hunk 11 above)
Comment (benironside) [reply docs/experimental-features/beaconing-detection.asciidoc:35]: good call.

File: docs/experimental-features/beaconing-detection.asciidoc [hunk 16]
```diff
@@ -0,0 +1,50 @@
...
+image::images/beaconing-detection-1.png[]
+
+[discrete]
+=== Feature details
+
+This feature uses a {ref}/transforms.html[transform] to segment network data by host and process name, then runs scripted metric aggregations on the host-process name pairs. For a given time window, the scripted metric aggregation checks each partition for the following:
+
+* Signals repeating at regular intervals, accounting for minor variations in those intervals.
+* Low variation of bytes sent from source to destination.
+* Low variation of bytes sent from destination to source.
```
Comment (jmikell821) [inline docs/experimental-features/beaconing-detection.asciidoc:23]: @ajosh0504 @SourinPaul I am slightly unsure about the language we are using here and just want to make sure I understand. What do we mean when we say: 
> "to segment network data by host/process name..."

Do we mean to filter? 

> "the scripted metric aggregation checks each partition for the following..."

What's a partition?  

Overall, I just want to make sure the language is simplistic and clear here.

File: docs/experimental-features/beaconing-detection.asciidoc [hunk 17]
```diff
@@ -0,0 +1,50 @@
...
+
+The transform, which runs every hour, also filters out common, known applications and IPs in an attempt to reduce false positives. The transform outputs information about the detection and the process and host indicators, for example:
+
+[role="screenshot"]
+image::images/beaconing-detection-2.png[]
+The values highlighted above are typical of beaconing behavior, and can help in your investigation.
```
Comment (jmikell821) [inline docs/experimental-features/beaconing-detection.asciidoc:29]:
```suggestion
The values highlighted above are typical of beaconing behavior and can help with your investigation.
```

File: docs/experimental-features/beaconing-detection.asciidoc [hunk 18]
```diff
@@ -0,0 +1,50 @@
...
+
+[discrete]
+=== Further customizations
+
+Advanced users can also tune the scripted metric aggregation's parameters, such as jitter percentage or time window. To overwrite the default parameters: delete the transform, change the parameters, and restart the transform. The configurable parameters are:
+
+* `number_buckets_in_range`: The number of time buckets into which the time window is split. Using more buckets will improve estimates for the various statistics, but also increase resource usage.
```
Comment (jmikell821) [inline docs/experimental-features/beaconing-detection.asciidoc:36]:
```suggestion
* `number_buckets_in_range`: The number of time buckets into which the time window is split. Using more buckets improves estimates for various statistics, but also increases resource usage.
```

File: docs/experimental-features/beaconing-detection.asciidoc [hunk 19]
```diff
@@ -0,0 +1,50 @@
...
+* `time_bucket_length`: The length of each time bucket. A higher value indicates a longer time window. Set this higher to check for very low-frequency beacons.
```
Comment (jmikell821) [inline docs/experimental-features/beaconing-detection.asciidoc:37]:
```suggestion
* `time_bucket_length`: The length of each time bucket. A higher value indicates a longer time window. Set this to a higher value to check for very low-frequency beacons.
```

File: docs/experimental-features/beaconing-detection.asciidoc [hunk 20]
```diff
@@ -0,0 +1,50 @@
...
+* `number_destination_ips`: The number of destination IPs to gather in the results. Setting this higher increases resource usage.
```
Comment (jmikell821) [inline docs/experimental-features/beaconing-detection.asciidoc:38]:
```suggestion
* `number_destination_ips`: The number of destination IPs to gather in results. Setting this to a higher value increases resource usage.
```

File: docs/experimental-features/beaconing-detection.asciidoc [hunk 21]
```diff
@@ -0,0 +1,50 @@
...
+* `max_beaconing_count_rv`: The maximum relative variance in the bucket counts for the high frequency beacon test. As with `max_beaconing_bytes_cov`, tuning this parameter involves a tradeoff between recall and precision.
+* `truncate_at`: The lower and upper fraction of bucket values discarded when computing `max_beaconing_bytes_cov` and `max_beaconing_count_rv`. This allows you to ignore occasional changes in traffic patterns. However, if you retain too small a fraction of the data, these tests will be unreliable.
+* `min_beaconing_count_autocovariance`: The minimum autocorrelation of the signal for the low-frequency beacon test. Lowering this value generally increases recall for malicious command and control beacons, while reducing precision.
+* `max_jitter`: The maximum amount of https://en.wikipedia.org/wiki/Jitter[jitter] assumed to be possible for a periodic beacon, as a fraction of its period.
+
+You can also make changes to the transform query. The default query looks for beaconing activity over a 6-hour time range, but you can use a different time range.
```
Comment (jmikell821) [inline docs/experimental-features/beaconing-detection.asciidoc:45]:
```suggestion
You can also make changes to the transform query. The default query looks for beaconing activity over a 6-hour time range, but you can change it.
```

File: docs/experimental-features/beaconing-detection.asciidoc [hunk 22]
```diff
@@ -0,0 +1,50 @@
...
+
+As mentioned previously, beaconing is not used exclusively by malware. Many legitimate, benign processes also exhibit beacon-like activity. In order to reduce false positives, a list of starter filters is included by default in the transform query to exclude known beaconing processes and IPs that fall into two groups:
```
Comment (jmikell821) [inline docs/experimental-features/beaconing-detection.asciidoc:47]:
```suggestion
Beaconing is not used exclusively by malware. Many legitimate, benign processes also exhibit beacon-like activity. To reduce false positives, a list of starter filters is included by default in the transform query to exclude known beaconing processes and IPs that fall into two groups:
```

**Review by jmikell821 — COMMENTED** [review]
@benironside round 1 of my feedback. I left a question for @ajosh0504 and @SourinPaul but in the meantime you can review my suggestions.

**Review by SourinPaul — APPROVED** [review]
Looks good pending the minor clarification you requested.

File: docs/experimental-features/beaconing-detection.asciidoc (hunk 16 above)
Comment (ajosh0504) [reply docs/experimental-features/beaconing-detection.asciidoc:23]: By segment we mean that we are splitting the data by host and process name. I don't think filter would mean the same?

Partition here is each pair of host-process name. You could replace "partition" with "pair" if that words better.

File: docs/experimental-features/beaconing-detection.asciidoc [hunk 23]
```diff
@@ -0,0 +1,50 @@
...
+
+* The source IP is local and the destination is remote.
+* The destination IP is in a block of known Microsoft IP addresses.
```
Comment (ajosh0504) [inline docs/experimental-features/beaconing-detection.asciidoc:50]: Can we also add a sentence mentioning that users can add to these filters based on what they see in their environment?

**Review by ajosh0504 — COMMENTED** [review]
Made some suggestions and @jmikell821 hopefully answered your questions.

File: docs/experimental-features/beaconing-detection.asciidoc [hunk 24]
```diff
@@ -0,0 +1,50 @@
...
+
+The release package includes dashboards for monitoring beaconing activity in your environment. You can review signals via a Lens dashboard called Network beaconing.
+
+[role="screenshot"]
+image::images/beaconing-detection-1.png[]
+
+[discrete]
+=== Feature details
+
+This feature uses a {ref}/transforms.html[transform] to segment network data by host and process name, then runs scripted metric aggregations on the host-process name pairs. For a given time window, the scripted metric aggregation checks each partition for the following:
```
Comment (jmikell821) [inline docs/experimental-features/beaconing-detection.asciidoc:19]:
```suggestion
This feature uses a {ref}/transforms.html[transform] to categorize network data by host and process name, then runs scripted metric aggregations on the host-process name pairs. For a given time window, the scripted metric aggregation checks each partition for the following:
```

**Comment by benironside** [comment]
@SourinPaul @ajosh0504 
I would appreciate one final review of the changes I made in response to feedback. Thanks!

**Review by ajosh0504 — APPROVED** [review]
LGTM!

File: docs/experimental-features/beaconing-detection.asciidoc [hunk 25]
```diff
@@ -0,0 +1,52 @@
...
+
+[role="screenshot"]
+image::images/beaconing-detection-1.png[]
+
+[discrete]
+=== Feature details
+
+This feature uses a {ref}/transforms.html[transform] to categorize network data by host and process name, then runs scripted metric aggregations on the host-process name pairs. For a given time window, the scripted metric aggregation checks each pair for the following:
+
+* Signals repeating at regular intervals, accounting for minor variations in those intervals.
```
Comment (nastasha-solomon) [inline docs/experimental-features/beaconing-detection.asciidoc:21]:
```suggestion
* Signals repeating at regular intervals, accounting for minor variations in those intervals
```

**Review by joepeeples — APPROVED** [review]
LGTM!

File: docs/experimental-features/beaconing-detection.asciidoc [hunk 26]
```diff
@@ -0,0 +1,52 @@
...
+The values highlighted above are typical of beaconing behavior and can help with your investigation.
+
+[discrete]
+=== Further customizations
+
+Advanced users can also tune the scripted metric aggregation's parameters, such as jitter percentage or time window. To overwrite the default parameters: delete the transform, change the parameters, and restart the transform. The configurable parameters are:
+
+* `number_buckets_in_range`: The number of time buckets into which the time window is split. Using more buckets improves estimates for various statistics, but also increases resource usage.
+* `time_bucket_length`: The length of each time bucket. A higher value indicates a longer time window. Set this to a higher value to check for very low-frequency beacons.
+* `number_destination_ips`: The number of destination IPs to gather in results. Setting this to a higher value increases resource usage.
```
Comment (nastasha-solomon) [inline docs/experimental-features/beaconing-detection.asciidoc:38]:
```suggestion
* `number_destination_ips`: The number of destination IPs to collect results. Setting this to a higher value increases resource usage.
```

**Review by nastasha-solomon — APPROVED** [review]
Just a handful of minor comments. Otherwise, LGTM!

File: docs/experimental-features/beaconing-detection.asciidoc (hunk 25 above)
Comment (joepeeples) [reply docs/experimental-features/beaconing-detection.asciidoc:21]: I feel like I've looked for guidelines on this and haven't found anything, and I've probably seen it both ways in Elastic docs. IMO periods are OK for longer phrases like these.

File: docs/experimental-features/beaconing-detection.asciidoc [hunk 27]
```diff
@@ -0,0 +1,52 @@
...
+[discrete]
+=== Feature details
+
+This feature uses a {ref}/transforms.html[transform] to categorize network data by host and process name, then runs scripted metric aggregations on the host-process name pairs. For a given time window, the scripted metric aggregation checks each pair for the following:
+
+* Signals repeating at regular intervals, accounting for minor variations in those intervals.
+* Low variation of bytes sent from source to destination.
+* Low variation of bytes sent from destination to source.
+
+The transform, which runs every hour, also filters out common, known applications and IPs in an attempt to reduce false positives. The transform outputs information about the detection and the process and host indicators, for example:
```
[the remark this answers is not in the document]
Comment (SourinPaul) [reply docs/experimental-features/beaconing-detection.asciidoc:25]: _detection_ is used generically here and does not refer to an Elastic Detection Rule/ Alert.

**Review by SourinPaul — APPROVED** [review]
lgtm

File: docs/experimental-features/beaconing-detection.asciidoc (hunk 25 above)
Comment (jmikell821) [reply docs/experimental-features/beaconing-detection.asciidoc:21]: Yes, to what Joe said ^.

File: docs/experimental-features/beaconing-detection.asciidoc (hunk 27 above)
Comment (jmikell821) [inline docs/experimental-features/beaconing-detection.asciidoc:25]:
```suggestion
The transform, which runs every hour, also filters out common, known applications and IPs attempt to reduce false positives. The transform outputs information about the detection and process and host indicators, for example:
```

File: docs/experimental-features/beaconing-detection.asciidoc [hunk 28]
```diff
@@ -0,0 +1,52 @@
...
+
+* `number_buckets_in_range`: The number of time buckets into which the time window is split. Using more buckets improves estimates for various statistics, but also increases resource usage.
+* `time_bucket_length`: The length of each time bucket. A higher value indicates a longer time window. Set this to a higher value to check for very low-frequency beacons.
+* `number_destination_ips`: The number of destination IPs to gather in results. Setting this to a higher value increases resource usage.
+* `max_beaconing_bytes_cov`: The maximum coefficient of variation in the payload bytes for the low source and destination bytes variance test. Higher values increase the chance of flagging traffic as beaconing, increasing https://en.wikipedia.org/wiki/Precision_and_recall[recall] while reducing https://en.wikipedia.org/wiki/Precision_and_recall[precision].
+* `max_beaconing_count_rv`: The maximum relative variance in the bucket counts for the high frequency beacon test. As with `max_beaconing_bytes_cov`, tuning this parameter involves a tradeoff between recall and precision.
```
Comment (jmikell821) [inline docs/experimental-features/beaconing-detection.asciidoc:40]:
```suggestion
* `max_beaconing_count_rv`: The maximum relative variance in the bucket counts for the high-frequency beacon test. As with `max_beaconing_bytes_cov`, tuning this parameter involves a tradeoff between recall and precision.
```

**Review by jmikell821 — COMMENTED** [review]
Few small comments.
